### DISCUSSION TITLE
Any way to apply centeredScaling on only X axis???

### DISCUSSION THREAD

**Discussion by kartiksolanki** [fabricjs/fabric.js#7944] [Q&A] [answered]
I am looking for a way to apply centered scaling on X axis and free scaling on Y axis. How do I achieve that???? 

Or Is there any way I can check whether horizontal scaling is being performed or vertical scaling is being performed on object:scaling event ??

**Comment by ShaMan123** [comment]
You should refactor the control action
This is where I would start looking
https://github.com/fabricjs/fabric.js/blob/23129ed6199d4fff21ccd1ae896b17b802cae5b9/src/controls.actions.js#L628

**Comment by kartiksolanki** [comment]
How should I edit that and I would love some help in understanding that core code.

**Reply by kartiksolanki** [reply]
Here is my jsfiddle link: http://jsfiddle.net/nexc2q8t/20/

**Comment by kartiksolanki** [comment, marked as answer]
Never mind, found the solution after digging more. Fabric js have **obj.__corner** property and I was able to enable/disable centeredScaling based on obj.__corner control selection inside **mouse:down:before** event.
